src/main/java/frc/robot/swerve/ModuleIOReal.java

```diff
@@ -0,0 +1,181 @@
+package frc.robot.swerve;
+
+import com.ctre.phoenix6.configs.Slot0Configs;
+import com.ctre.phoenix6.configs.TalonFXConfiguration;
+import com.ctre.phoenix6.controls.*;
+import com.ctre.phoenix6.hardware.TalonFX;
+import com.ctre.phoenix6.signals.NeutralModeValue;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.SwerveModulePosition;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.DutyCycleEncoder;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.RunCommand;
+import utils.math.AngleUtil;
+import utils.math.differential.Integral;
+
+public class ModuleIOReal implements ModuleIO {
+
+    private final TalonFX driveMotor;
+    private final TalonFX angleMotor;
+    private final DutyCycleEncoder encoder;
+    private final Integral driveSupplyChargeUsedCoulomb = new Integral();
+    private final Integral driveStatorChargeUsedCoulomb = new Integral();
+    private final Integral angleSupplyChargeUsedCoulomb = new Integral();
+    private final Integral angleStatorChargeUsedCoulomb = new Integral();
+    private double angleSetpoint;
+    private double currentAngle;
+    private double driveMotorVelocitySetpoint;
+
+    public ModuleIOReal(int driveMotorID, int angleMotorID, int encoderID,
+                        double[] motionMagicConfigs, int number) {
+        this.driveMotor = new TalonFX(driveMotorID);
+        this.angleMotor = new TalonFX(angleMotorID);
+
+        this.encoder = new DutyCycleEncoder(encoderID);
+
+        TalonFXConfiguration driveMotorConfig = new TalonFXConfiguration();
+        Slot0Configs drivePIDGains = new Slot0Configs()
+                .withKP(SwerveConstants.DRIVE_kP)
+                .withKI(SwerveConstants.DRIVE_kI)
+                .withKD(SwerveConstants.DRIVE_kD)
+                .withKV(SwerveConstants.DRIVE_KF);
```

Comment:
What I meant is to save `drivePIDGains` in the constants file (don't forget to use `CAPITAL_LETTERS`). You can get rid of the individual PID gains:
```java
// Constants file:
public static final Slot0Configs = new Slot0Configs().withKP(1.1).withKI(2.2).withKP(3.3).withKV(4.4);
```